src/main/java/org/spongepowered/api/text/dictionary/Dictionary.java

```diff
@@ -0,0 +1,394 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.text.dictionary;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.Text.Placeholder;
+import org.spongepowered.api.text.Texts;
+import org.spongepowered.api.text.Texts.FormatBuilder;
+import org.spongepowered.api.text.format.TextColors;
+import org.spongepowered.api.util.BiFunctions;
+import org.spongepowered.api.util.Functions;
+import org.spongepowered.api.util.command.CommandSource;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * Utility class to allow plugin developers to easily add multi language support
+ * to their plugins.
+ */
+public class Dictionary implements Function<String, Function<Locale, Text>> {
+
+    private static final BiFunction<Locale, Map<Locale, Text>, Text> TRANSLATION_FALLBACK = (locale, map) -> map.values().iterator().next();
+    private final Map<String, Map<Locale, Text>> messages = new HashMap<>();
+    private final Function<String, String> messageFallback;
+    private final BiFunction<Locale, Map<Locale, Text>, Text> translationFallback;
+
+    /**
+     * Creates a new dictionary with no message fallback and a first result
+     * translation fallback.
+     */
+    public Dictionary() {
+        this(Functions.constantNull(), BiFunctions.constantNull());
+    }
+
+    /**
+     * Creates a new dictionary with the given message and translation fallback.
+     *
+     * @param messageFallback The message fallback function to use if the
+     *        requested translaion id could not be found
+     * @param translationFallback The translation fallback to use if the
+     *        requested there is no translation for the given language, but
+     *        translations for other languages
+     */
+    public Dictionary(Function<String, String> messageFallback,
+            BiFunction<Locale, Map<Locale, Text>, Text> translationFallback) {
+        super();
```

Comment:
Are you referring to the `super()` call?
This was generated automatically and I didn't bother to remove it yet.